internal/core/writer/properties.go

```diff
@@ -59,39 +59,45 @@ func newDimensionPropertyClient(conf *config.WriterConfig) *dimensionPropertyCli
 				TLSHandshakeTimeout: 10 * time.Second,
 			},
 		},
-		history: history,
-		reqSema: make(chan struct{}, int(conf.PropertiesMaxRequests)),
+		history:           history,
+		reqSema:           make(chan struct{}, int(conf.PropertiesMaxRequests)),
+		PropertyFilterSet: conf.PropertyFilter,
 	}
 }
 
 // SetPropertiesOnDimension will set custom properties on a specific dimension
 // value.  It will wipe out any description or tags on the dimension.  There is
 // no retry logic here so any failures are terminal.
 func (dpc *dimensionPropertyClient) SetPropertiesOnDimension(dimProps *types.DimProperties) error {
-	if !dpc.isDuplicate(dimProps) {
-		dpc.reqSema <- struct{}{}
+	filteredDimProps := &(*dimProps)
 
-		log.WithFields(log.Fields{
-			"name":  dimProps.Name,
-			"value": dimProps.Value,
-			"props": dimProps.Properties,
-			"tags":  dimProps.Tags,
-		}).Info("Syncing properties to dimension")
+	filteredDimProps = applyFilter(filteredDimProps, dpc.PropertyFilterSet)
+	if filteredDimProps == nil {
+		return nil
+	}
 
-		err := dpc.doReq(dimProps.Name, dimProps.Value, dimProps.Properties, dimProps.Tags)
+	if !dpc.isDuplicate(filteredDimProps) {
+		dpc.reqSema <- struct{}{}
+		err := dpc.doReq(filteredDimProps.Name, filteredDimProps.Value,
+			filteredDimProps.Properties, filteredDimProps.Tags)
 		<-dpc.reqSema
 		if err != nil {
 			return err
 		}
 		// Add it to the history only after successfully propagated.  This
 		// could lead to some duplicates if there are multiple concurrent calls
 		// for the same dim props, but that's ok.
-		dpc.history.Add(dimProps.Dimension, dimProps)
+		dpc.history.Add(filteredDimProps.Dimension, filteredDimProps)
 		atomic.AddInt64(&dpc.TotalPropUpdates, int64(1))
 	}
 	return nil
 }
 
+// applyFilter filters out properties from filter sets and returns
+func applyFilter(dimProps *types.DimProperties, propFilters *propfilters.FilterSet) *types.DimProperties {
```

Comment:
Get rid of this function and just do the equivalent in line 74.  This is unnecessary redirection that just makes it harder to understand.